src/cloudcasting/download.py

```diff
@@ -129,6 +144,9 @@ def download_satellite_data(
 
     years = range(start_date_stamp.year, end_date_stamp.year + 1)
 
+    # Create a list of dates to download
+    dates_to_download = pd.date_range(start_date_stamp, end_date_stamp, freq=download_frequency)
```

Comment:
I think there is one last gotcha here. If `start_date_stamp` is not a multiple of `download_frequency` then it will cause issues. For example if `start_date_stamp="2022-01-01 12:05"` and `download_frequency="15min"` then the date range function would return `["2022-01-01 12:05",  "2022-01-01 12:20", "2022-01-01 12:35", ...]`. 

I think the desired behaviour in this instance would be for it to return `["2022-01-01 12:15",  "2022-01-01 12:30", "2022-01-01 12:45", ...]`. 

I think we can do this by using `pd.date_range(start_date_stamp.ceil(download_frequency), end_date_stamp, freq=download_frequency)`

Reply:
Nice catch! I'm learning lots of things about datetime utils here — can confirm this works as you've stated from my testing. Will implement!